packages/noya-state/src/selectors/pointSelectors.ts

```diff
@@ -84,3 +97,209 @@ export const getSelectedControlPoint = (
 
   return controlPoints.length > 0 ? controlPoints[0] : undefined;
 };
+
+export const getIsEditingPath = (type: InteractionState['type']): boolean => {
+  return (
+    type === 'editPath' ||
+    type === 'maybeMovePoint' ||
+    type === 'movingPoint' ||
+    type === 'maybeMoveControlPoint' ||
+    type === 'movingControlPoint'
+  );
+};
+
+export const moveSelectedPoints = (
+  selectedPointLists: SelectedPointLists,
+  layerIndexPaths: IndexPath[],
+  delta: Partial<Point>,
+  mode: SetNumberMode,
+  draftPage: Draft<Sketch.Page>,
+  pageSnapshot: Sketch.Page,
+  CanvasKit: CanvasKit,
+) => {
+  const boundingRects = getBoundingRectMap(
+    pageSnapshot,
+    Object.keys(selectedPointLists),
+    {
+      clickThroughGroups: true,
+      includeArtboardLayers: false,
+      includeHiddenLayers: false,
+    },
+  );
+
+  layerIndexPaths.forEach((indexPath) => {
+    const layer = Layers.access(pageSnapshot, indexPath);
+    const pointList = selectedPointLists[layer.do_objectID];
+    const boundingRect = boundingRects[layer.do_objectID];
+
+    if (!Layers.isPointsLayer(layer) || !boundingRect) return;
+
+    // Update all points by first transforming to the canvas's coordinate system
+    const decodedPoints = layer.points
+      .map((curvePoint) => decodeCurvePoint(curvePoint, boundingRect))
+      .map((decodedPoint, index) => {
+        if (!pointList.includes(index)) return decodedPoint;
+
+        (['point', 'curveFrom', 'curveTo'] as const).forEach((key) => {
+          decodedPoint[key] = {
+            x:
+              delta.x === undefined
+                ? decodedPoint[key].x
+                : mode === 'replace'
+                ? delta.x
+                : decodedPoint[key].x + delta.x,
+            y:
+              delta.y === undefined
+                ? decodedPoint[key].y
+                : mode === 'replace'
+                ? delta.y
+                : decodedPoint[key].y + delta.y,
+          };
+        });
+
+        return decodedPoint;
+      });
+
+    // Determine the new bounds of the updated points
+    const [minX, minY, maxX, maxY] = path(
+      CanvasKit,
+      decodedPoints.map((decodedCurvePoint) =>
+        encodeCurvePoint(decodedCurvePoint, layer.frame),
+      ),
+      layer.frame,
+    ).computeTightBounds();
+
+    const newRect: Rect = {
+      x: minX,
+      y: minY,
+      width: maxX - minX,
+      height: maxY - minY,
+    };
+
+    const draftLayer = Layers.access(draftPage, indexPath) as PointsLayer;
+
+    draftLayer.frame = {
+      ...layer.frame,
+      ...newRect,
+    };
+
+    // Transform back to the range [0, 1], using the new bounds
+    draftLayer.points = decodedPoints.map((decodedCurvePoint) =>
+      encodeCurvePoint(decodedCurvePoint, draftLayer.frame),
+    );
+  });
+};
```

Comment:
Do we wanna try to make this a separate function? It's ok if we leave it for another time too